src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -26,8 +28,58 @@ class TransposeValueTensors : public ov::pass::MatcherPass {
         using Ref = std::reference_wrapper<Context>;
     };
 
-    OPENVINO_MATCHER_PASS_RTTI("npuw::LLMCompiledModel::TransposeValueTensors");
-    TransposeValueTensors(Context::Ref ctx) {
+protected:
+    // generic part of matchers, to transpose v-tensors, and concat, and update matmul args 
+    void transpose_matmul_b(Context::Ref ctx, 
+            std::shared_ptr<ov::Node> node_param,
+            std::shared_ptr<ov::Node> node_concat,
+            std::shared_ptr<ov::Node> node_transpose,
+            std::shared_ptr<ov::Node> node_matmul) {
+
+        auto matched_param = std::static_pointer_cast<ov::op::v0::Parameter>(node_param);
+        auto matched_concat = std::static_pointer_cast<ov::op::v0::Concat>(node_concat);
+        auto matched_transpose = std::static_pointer_cast<ov::op::v1::Transpose>(node_transpose);
+        auto matched_matmul = std::static_pointer_cast<ov::op::v0::MatMul>(node_matmul);
+
+        auto param_shape = matched_param->get_partial_shape();
+        OPENVINO_ASSERT(param_shape.size() == 4u);
+        // NB: Transpose Parameter that correspond to V-tensor it will
+        // speed-up its multiplication with attention scores
+        std::swap(param_shape[2], param_shape[3]);
+        auto new_param = std::make_shared<ov::opset13::Parameter>(matched_param->get_element_type(), param_shape);
+        new_param->set_friendly_name(matched_param->get_friendly_name());
+        new_param->outputs().begin()->get_tensor().set_names(
+            matched_param->outputs().begin()->get_tensor().get_names());
+        ov::replace_node(matched_param, new_param);
+        // NB: Save in order to add/remove to the model later on
+        ctx.get().new_params.push_back(new_param);
+        ctx.get().old_params.push_back(matched_param);
+
+        auto order_cst = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{4}, {0, 2, 3, 1});
```

Comment:
is it enough to replace only the "order" constant? 
it looks like, the new subgraph is the same as the previous one except the parameter shape and order_cst

Reply:
@itikhono kind of, subgraph is same except of transposed tensor, so changes propagated this to 2 more constants comparing to llama2 case.

Reply:
@itikhono I've checked both proposed solutions on unit tests - looks it is working, will update on upcoming PR, thanks. This code actually was a legacy , invented before this PR, so also confused me a lot.